internal/ascpsettlement/postgres.go

```
@@ -0,0 +1,760 @@
+	query := `
+		SELECT transaction_id, kind
+		FROM ascp_ledger_transactions original
+		WHERE operation_id=$1 AND kind <> 'REVERSAL'
+		  AND NOT EXISTS (SELECT 1 FROM ascp_ledger_transactions reversal WHERE reversal.reversal_of=original.transaction_id)`
+	args := []any{operation.OperationID}
+	if onlyKind != "" {
+		query += ` AND kind=$2`
+		args = append(args, onlyKind)
+	}
+	query += ` ORDER BY recorded_at, transaction_id FOR UPDATE`
+	rows, err := tx.QueryContext(ctx, query, args...)
```

golangci-lint (2.12.2)

[error] rows.Err must be checked

(rowserrcheck)